app/services/search/classified_listing.rb

```diff
@@ -0,0 +1,139 @@
+          index: INDEX_ALIAS,
+          body: serialized_data,
+        )
+      end
+
+      def find_document(classified_listing_id)
+        SearchClient.get(id: classified_listing_id, index: INDEX_ALIAS)
+      end
+
+      def create_index(index_name: INDEX_NAME)
+        SearchClient.indices.create(index: index_name, body: settings)
+      end
+
+      def delete_index(index_name: INDEX_NAME)
+        SearchClient.indices.delete(index: index_name)
+      end
+
+      def add_alias(index_name: INDEX_NAME, index_alias: INDEX_ALIAS)
+        SearchClient.indices.put_alias(index: index_name, name: index_alias)
+      end
+
+      def update_mappings(index_alias: INDEX_ALIAS)
+        SearchClient.indices.put_mapping(index: index_alias, body: mappings)
+      end
+
+      private
+
+      def settings
+        { settings: { index: index_settings } }
+      end
+
+      def index_settings
+        if Rails.env.production?
+          {
+            number_of_shards: 1,
+            number_of_replicas: 1
+          }
+        else
+          {
+            number_of_shards: 1,
+            number_of_replicas: 0
+          }
+        end
+      end
+
+      def mappings
+        # 1. "In Elasticsearch, there is no dedicated array datatype. Any field can
+        #    contain zero or more values by default, however, all values in the
+        #    array must be of the same datatype."
+        #
+        # https://www.elastic.co/guide/en/elasticsearch/reference/current/array.html
+
+        # 2. You don't have to specify type: "object" since it is the default.
+        #    Specifying type: "object" will break specs because when you later
+        #    call the mappings API on Elasticsearch, it will NOT return the
+        #    type: object key, value pair.
+        #
+        # https://www.elastic.co/guide/en/elasticsearch/reference/current/object.html#object
+
+        {
+          dynamic: "strict",
+          properties: {
+            id: {
+              type: "keyword"
+            },
+            author: { # Don't need to specify type: "object" - see comment 2 above
+              dynamic: "strict",
+              properties: {
+                username: {
+                  type: "keyword"
+                },
+                name: {
+                  type: "keyword"
+                },
+                profile_image_90: {
+                  type: "keyword"
+                }
+              }
+            },
+            bumped_at: {
+              type: "date"
+            },
+            category: {
+              type: "keyword"
+            },
+            contact_via_connect: {
+              type: "boolean"
+            },
+            expires_at: {
+              type: "date"
+            },
+            location: {
+              type: "text",
+              fields: {
+                raw: {
+                  type: "keyword"
+                }
+              }
+            },
+            processed_html: {
```

Comment:
Here I want to be smarter with our fields than we are in Algolia. We search on this processed_html [text field](https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html#text) but the downside is that it is going to [get analyzed and contain a lot of "tokens"](https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html#text) we dont want to search like the html tags. If we need this for the view then we should store it as a keyword(unanalyzed) and add `body_markdown` to this mapping and search on that instead. I think the searches will be a lot more accurate.

Reply:
Love this! Great idea 💯